WRF.md

```diff
@@ -0,0 +1,204 @@
+    make -j 16
+    make install
+## Compiling and Installing WRF
+1. Run the following command to obtain the source code package:
+
+
+    wget https://github.com/wrf-model/WRF/archive/refs/tags/v4.2.tar.gz
+2. Decompress the WRF installation package:
+
+
+    tar -zxvf v4.2.tar.gz
+3. Run the following command to switch to WRF source code directory:
+
+
+    cd WRF-4.2
+4. Run the following commands to configure the pre-compilation environment:
+
+
+    export WRFIO_NCD_LARGE_FILE_SUPPORT=1
+    export NETCDF=/path/to/netcdf
+    export HDF5=/path/to/hdf5
+    export PNETCDF=/path/to/pnetcdf
+    export CPPFLAGS="-I$HDF5/include -I$PNETCDF/include -I$NETCDF/include"
+    export LDFLAGS="-L$HDF5/lib -L$PNETCDF/lib -L$NETCDF/lib -lnetcdf -lnetcdff -lpnetcdf -lhdf5_hl -lhdf5 -lz"
+5. Edit the arch/configure.defaults file and add the following information before line 1977:
+
+
+    ################################################## #########
+    #ARCH   Linux   aarch64,gnu OpenMPI #serial smpar dmpar dm+sm
+    DESCRIPTION     =       GNU ($SFC/$SCC)
+    DMPARALLEL      =        1
+    OMPCPP          =        -D_OPENMP
+    OMP             =        -fopenmp
+    OMPCC           =        -fopenmp
+    SFC             =       gfortran
+    SCC             =       gcc
+    CCOMP           =       gcc
+    DM_FC           =       mpif90 -f90=$(SFC)
+    DM_CC           =       mpicc -cc=$(SCC) -DMPI2_SUPPORT
+    FC              =       CONFIGURE_FC
+    CC              =       CONFIGURE_CC
+    LD              =       $(FC)
+    RWORDSIZE       =       CONFIGURE_RWORDSIZE
+    PROMOTION       =       #-fdefault-real-8
+    ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM
+    CFLAGS_LOCAL    =       -w -O3 -c -march=armv8.2-a
+    LDFLAGS_LOCAL   =
+    CPLUSPLUSLIB    =
+    ESMF_LDFLAG     =      $(CPLUSPLUSLIB)
+    FCOPTIM         =       -O3 -ftree-vectorize -funroll-loops -march=armv8.2-a
+    FCREDUCEDOPT    =       $(FCOPTIM)
+    FCNOOPT         =       -O0
+    FCDEBUG         =       # -g $(FCNOOPT)  # -fbacktrace -ggdb-fcheck=bounds,do,mem,pointer -ffpe-trap=invalid,zero,overflow
+    FORMAT_FIXED    =       -ffixed-form
+    FORMAT_FREE     =       -ffree-form -ffree-line-length-none
+    FCSUFFIX        =
+    BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+    FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
+    FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
+    MODULE_SRCH_FLAG =
+    TRADFLAG        =      -traditional
+    CPP             =      /lib/cpp -P
+    AR              =      ar
+    ARFLAGS         =      ru
+    M4              =      m4 -G
+    RANLIB          =      ranlib
+    RLFLAGS         =
+    CC_TOOLS        =      $(SCC)
+6. Run the following command to generate a configuration file:
+
+
+    echo 4 | ./configure
+    1->"enter"
```

Comment:
I don't understand this line.

Reply:
I was able to overcome this problem.
```
################################################## #########
#ARCH   Linux   aarch64,gnu OpenMPI #serial smpar dmpar dm+sm
```
needs a third line with a single `#` in it:
```
################################################## #########
#ARCH   Linux   aarch64,gnu OpenMPI #serial smpar dmpar dm+sm
#
```

With this `./configure` works and asks me 2 questions. I used `4` as an answer for the first and `1` for the second. I will reword the documentation!

Reply:
https://github.com/Yikun/hpc-arm/commit/b3516f1a39c34248a143e8e9c54546fcc9ef1d44